### DISCUSSION TITLE
Problem in ImGui::NewFrame CRASH when calls internal GetDefaultFont()

### DISCUSSION THREAD

**Discussion by RodrigoBrombatti** [ocornut/imgui#6063] [New Users Build/Link/Run issues ONLY!]
Version: 1.89.2
Branch: master

Back-ends: custom (GLFW3 + GLAD)
Compiler: Clang
Operating System: MacOS

Trying to follow The Chermo game engine tutorial but I can't make ImGui work

Im trying to follow the chermo tutorial, I noticed that he was using old version of ImGui.
So in the newer I get I need to setIMGUI_IMPL_OPENGL_LOADER_CUSTOM to load my custom GL loader with Glad etc
Im building ImGui and I can see the GLFW OpenGL3 examples working well
So, I think I did every initial setup correctly and Im adding ImGui to my personal engine lib with Add_Subdirectory and linking that correctly. Im about 4 days now trying to fix it or something I did wrong. 

I follow every step in Internal new frame and I see ImGui IO getting every info correctly, until this reaches GetDefaultFont() from imgui_internal.h

I got this:
EXC_BAD_ACCESS (code=1, address=0x68)

Everyone tells me is something bad in my code but I review everithing like many many times and im going crazy 
Maybe is a newbie error im doing here but im completely stuck so help me please

My Repo Complete:
https://github.com/RodrigoBrombatti/NuloEngine

GL Loader: (just like copy and paste of backends/imgui_impl_opengl3.h and .cpp with glad added)
https://github.com/RodrigoBrombatti/NuloEngine/blob/master/Nulo/src/Nulo/Platform/OpenGL/ImGuiOpenGLRenderer.cpp

Trying to load:
https://github.com/RodrigoBrombatti/NuloEngine/blob/master/Nulo/src/Nulo/ImGui/ImGuiLayer.cpp

What I did to build ImGui:
1 - add submodule
2 - got this CMakeLists.txt for Build image
https://gist.github.com/rokups/f771217b2d530d170db5cb1e08e9a8f4
1 - mkdir build
2 - cmake .. inside imgui with CMakeLists.txt
3 - cmake --build . inside vendor/imgui/build
4 - target my engine lib with imgui and include_directories(vendor/imgui/)

also, my principal CMakeLists.txt
https://github.com/RodrigoBrombatti/NuloEngine/blob/master/CMakeLists.txt

Im using:
Apple MacBook M1 Pro with Apple Silicon (ARM64)

If someone can help me because I don't know what to do anymore

Thanks and have a great new year

**Comment by PathogenDavid** [comment]
Your repo is private so we can't see it.

I suspect you're calling something in the wrong order. Crashes there implies to me that you might be calling `ImGui::NewFrame()` before the `_NewFrame` functions for the backends. (IE: `ImGui_ImplOpenGL3_NewFrame`/`ImGui_ImplGlfw_NewFrame`.)

Ensure you're running a debug build. You shouldn't be able to get this far without an assert going bang and giving you some info about what you're doing wrong.

Carefully compare your code with [`example_glfw_opengl3`](https://github.com/ocornut/imgui/blob/d7c8516a4b848c0291e3d75b627c0843f515f591/examples/example_glfw_opengl3/main.cpp) and ensure you're doing things in logically the same order.

It'd be a good idea to run both the example and your code at the same time in a debugger and step through each one's initialization one at function at a time to ensure things are happening in the same order. (Or put breakpoints on relevant function calls in each and make sure they're hit in the same order.)

> So in the newer I get I need to setIMGUI_IMPL_OPENGL_LOADER_CUSTOM to load my custom GL loader with Glad etc

Ideally you should just let the backend use the embedded loader. IE: Don't use `IMGUI_IMPL_OPENGL_LOADER_CUSTOM`.

`IMGUI_IMPL_OPENGL_LOADER_CUSTOM` is only meant for single-file unity builds.

You can still use Glad for your code.

**Reply by RodrigoBrombatti** [reply]
Sorry about that, I found the solution
Just my fault messing with static and dynamic libs

**Reply by PathogenDavid** [reply]
No worries, glad you were able to figure it out!

**Comment by RodrigoBrombatti** [comment]
Solved

**Reply by EinOnly** [reply]
How to solve it?
